build/notebooks/table_variants_missense.md

```diff
@@ -5070,7 +5070,246 @@ tbl_variants_selected.displayall(tr_style=tr_style, td_styles=td_styles)
+# \multicolumn{2}{c}{LD ($D'$)} \\
+# \cmidrule(r){1-3}
+# \cmidrule(r){4-12}
+# \cmidrule(r){13-14}
+# Position\tablefootnote{Position relative to AgamP3 reference sequence, chromosome arm 2L.} & 
+# \emph{Ag}\tablefootnote{Codon numbering according to transcript \texttt{AGAP004707-RA} in geneset AgamP4.4.} & 
+# \emph{Md}\tablefootnote{Codon numbering according to @@TODO.} & 
+# AO\emph{Ac} & 
+# BF\emph{Ac} & 
+# GN\emph{Ag} & 
+# BF\emph{Ag} & 
+# CM\emph{Ag} & 
+# GA\emph{Ag} & 
+# UG\emph{Ag} & 
+# KE & 
+# GW & 
+# \texttt{L995F} & 
+# \texttt{L995S} \\
+# \midrule
+# """
+# template = r"""
+# {substitution} & {AGAP004707-RA} & NA & {AF_AOM} & {AF_BFM} & {AF_GNS} & {AF_BFS} & {AF_CMS} & {AF_GAS} & {AF_UGS} & {AF_KES} & {AF_GWA} & NA & NA \\
+# """
+# epilogue = r"""
+# \bottomrule
+# \end{tabular}
+# """
+# tbl_variants_display.totext('../tables/variants_missense.tex', 
+#                             encoding='ascii',
+#                             prologue=prologue, 
+#                             template=template,
+#                             epilogue=epilogue)
+```
+
+
+```python
+# !cat ../tables/variants_missense.tex
+```
+
+# add housefly numbering
+
+
+```python
+#to easily split the codon numbers from the codon letters
+import re
+```
+
+
+```python
+#load the codon map from the blog post (with the header info removed)
+md_tbl = etl.fromtsv('/home/chris/Git/agam-vgsc-report/data/domestica_gambiae_map.txt')
+
+#get codons
+dom = list(md_tbl['domestica_codon'])
+ano = list(md_tbl['gambiae_codon'])
+```
+
+
+```python
+#make a dictionary with a as the key and d as the value
+map_dict = {a: d for a, d in zip(ano, dom)}
+```
+
+
+```python
+#get the snpeff annotations
+gam_cod = list(tbl_variants_selected['SNPEFF_HGVS_p'])
+```
+
+
+```python
+#ditch the codon letters - using re the regex module - LOVELY
+gam_cod_cl = []
+r = re.compile("([a-zA-Z]+)([0-9]+)")
+for c in gam_cod:
+    if c:
+        d = c[2:10]
+        m = r.match(d)
+        g = m.group(2)
+        gam_cod_cl.append(g)
+len(gam_cod_cl)
+```
+
+
+
+
+    20
+
+
+
+
+```python
+MD = [map_dict[c] for c in gam_cod_cl]
+MD
+```
+
+
 
+
+    ['261',
+     '410',
+     '.',
+     '508',
+     '810',
+     '1014',
+     '1014',
+     '1133',
+     '1262',
+     '1532',
+     '1575',
+     '1602',
+     '1608',
+     '1751',
+     '1858',
+     '1873',
+     '1879',
+     '1879',
+     '1939',
+     '1945']
+
+
+
+
+```python
+#get the musca codon letter at these positions and add
+fs = pyfasta.Fasta('/home/chris/Git/agam-vgsc-report/data/domestica_gambiae_PROT_MEGA.fas')
+```
+
+
+```python
+#grab the right sample
+dom = fs.get('domestica_vgsc')
+```
+
+
+```python
+#remove the '-' from the aligned fasta so the numbering makes sense
+dom_fix = [p for p in dom if p != '-']
+#check
+dom_fix[261-1],dom_fix[1945-1]
+```
+
+
+
+
+    ('R', 'I')
+
+
+
+
+```python
+#add them to the position
+```
+
+
+```python
+MD_fix = []
+for p in MD:
+    if p == '.':
+        MD_fix.append(p)
```

Comment:
I'm being fussy, but could you convert this to a '-' instead of a '.' here? I think '-' would look better in the final table, as it is the character used to symbolise a deletion in the protein alignment, hence the sense should be a bit clearer.